src/bp-core/profile-search/bps-xprofile.php

```diff
@@ -459,17 +459,22 @@ function bp_ps_xprofile_gender_users_search( $f ) {
 	if ( isset( $gender ) && ! empty( $gender ) ) {
 
 		$exists_gender = $wpdb->get_results( "SELECT COUNT(*) as count, id FROM {$table_profile_fields} a WHERE parent_id = 0 AND type = 'gender' " );
-
-		$custom_ids = $wpdb->get_col( "SELECT user_id FROM {$table_profile_data} WHERE field_id = {$exists_gender[0]->id} AND value = '{$gender}'" );
-
-		if ( isset( $custom_ids ) && ! empty( $custom_ids ) ) {
-			return $custom_ids;
-		} else {
-			return array();
+		if ( ! empty( $exists_gender ) ) {
+			$where = " AND value = '{$gender}'";
```

Comment:
@mahedihasannoman What if `$gender` is ARRAY. I think the condition should be in else section.

Reply:
Yes it could be ARRAY so added else condition there.